libpysal/graph/base.py

```diff
@@ -2651,6 +2651,10 @@ def describe(
         if (y.index != self.unique_ids).all():
             raise ValueError("The values index is not aligned with the graph index.")
 
+        # reset numerical index to enable numba functionality
+        if isinstance(y.index.dtype, object):
```

Comment:
```suggestion
        if not isinstance(y.index.dtype, int | float):
```
this is probably safer. you can also have geometry dtype and other so I'd explicitly check for those that work (do both ints and floats work?)